core-backend/wasmtime/src/funcs.rs

```diff
@@ -284,6 +313,36 @@ impl<E: Ext + 'static> FuncsHandler<E> {
         Func::wrap(store, func)
     }
 
+    pub fn reply_commit_wgas(store: &mut Store<StoreData<E>>, mem: WasmtimeMemory) -> Func {
+        let func = move |mut caller: Caller<'_, StoreData<E>>,
+                         gas_limit: u64,
+                         value_ptr: i32,
+                         message_id_ptr: i32| {
+            let ext = caller.data().ext.clone();
+            ext.with(|ext: &mut E| -> Result<i32, String> {
+                let mem_wrap = get_caller_memory(&mut caller, &mem);
+                let value = get_u128(&mem_wrap, value_ptr as usize)?;
+                ext.reply_commit(ReplyPacket::new_with_gas(
+                    Default::default(),
+                    gas_limit,
+                    value,
+                ))
+                .on_success_code(|message_id| {
+                    write_to_caller_memory(
+                        &mut caller,
+                        &mem,
+                        message_id_ptr as isize as _,
+                        message_id.as_ref(),
+                    )
+                })
+            })
+            .map_err(Trap::new)?
+            .map_err(|_| "Trapping: unable to send message")
```

Comment:
Recheck all trap explanations for other places please.

```suggestion
            .map_err(|_| "Trapping: unable to send reply message")
```